File: declarative_sqlite/lib/src/builders/key_builder.dart
import 'package:declarative_sqlite/src/schema/db_key.dart';

class KeyBuilder {
  final List<String> columns;
  KeyType _type = KeyType.indexed;

  KeyBuilder(this.columns);

  void primary() {
    _type = KeyType.primary;
  }

  void index() {
    _type = KeyType.indexed;
  }

  void unique() {
    _type = KeyType.unique;
  }

  DbKey build() {
    return DbKey(
      columns: columns,
      type: _type,
    );
  }
}

File: packages/core/src/schema/builders/key-builder.ts
import type { DbKey, KeyType } from '../types';

/**
 * Builder for database keys (primary, unique, index)
 */
export class KeyBuilder {
  private _type: KeyType = 'INDEX';
  private _name?: string;
  
  constructor(private columns: string[]) {}
  
  /**
   * Mark as primary key
   */
  primary(): this {
    this._type = 'PRIMARY';
    return this;
  }
  
  /**
   * Mark as unique constraint
   */
  unique(): this {
    this._type = 'UNIQUE';
    return this;
  }
  
  /**
   * Mark as index (default)
   */
  index(): this {
    this._type = 'INDEX';
    return this;
  }
  
  /**
   * Set custom name for the key
   */
  name(name: string): this {
    this._name = name;
    return this;
  }
  
  /**
   * Build the key definition
   */
  build(): DbKey {
    return {
      columns: this.columns,
      type: this._type,
      name: this._name,
    };
  }
}
